checkov/common/checks_infra/solvers/attribute_solvers/intersects_attribute_solver.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from typing import Optional, Any, Dict
+from collections.abc import Iterable
+from checkov.common.graph.checks_infra.enums import Operators
+from checkov.common.checks_infra.solvers.attribute_solvers.base_attribute_solver import BaseAttributeSolver
+
+logger = logging.getLogger(__name__)
+
+
+class IntersectsAttributeSolver(BaseAttributeSolver):
+    operator = Operators.INTERSECTS  # noqa: CCE003  # a static attribute
+
+    def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:  # type:ignore[override]
+        attr = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
+
+        if isinstance(self.value, str) and isinstance(attr, Iterable):
+            return self.value in attr
+
+        if type(self.value) in [list, set, dict] and isinstance(attr, Iterable):
```

Comment:
```suggestion
        if isinstance(self.value, Collection) and isinstance(attr, Iterable):
```
I think it is a `Collection` otherwise `Sequence` for sure 😄